test/k8sT/Policies.go

```diff
@@ -819,32 +843,40 @@ var _ = Describe("K8sPolicyTest", func() {
 				}
 			}
 
-			It("Tests proxy visibility without policy", func() {
-				checkProxyRedirection(app1PodIP, false)
+			proxyVisibilityTest := func(resource, podToAnnotate, anno string, parserType policy.L7ParserType) {
+				checkProxyRedirection(resource, false, parserType)
 
-				By("Annotating %s with <Ingress/80/TCP/HTTP>", app1Pod)
-				res := kubectl.Exec(fmt.Sprintf("%s annotate pod %s -n %s %s=\"<Ingress/80/TCP/HTTP>\"", helpers.KubectlCmd, app1Pod, namespaceForTest, annotation.ProxyVisibility))
+				By("Annotating %s with %s", podToAnnotate, anno)
+				res := kubectl.Exec(fmt.Sprintf("%s annotate pod %s -n %s %s=\"%s\"", helpers.KubectlCmd, podToAnnotate, namespaceForTest, annotation.ProxyVisibility, anno))
 				res.ExpectSuccess("annotating pod with proxy visibility annotation failed")
 				Expect(kubectl.CiliumEndpointWaitReady()).To(BeNil())
 
-				checkProxyRedirection(app1PodIP, true)
+				checkProxyRedirection(resource, true, parserType)
 
-				By("Removing proxy visibility annotation on %s", app1Pod)
-				kubectl.Exec(fmt.Sprintf("%s annotate pod %s -n %s %s-", helpers.KubectlCmd, app1Pod, namespaceForTest, annotation.ProxyVisibility)).ExpectSuccess()
+				By("Removing proxy visibility annotation on %s", podToAnnotate)
+				kubectl.Exec(fmt.Sprintf("%s annotate pod %s -n %s %s-", helpers.KubectlCmd, podToAnnotate, namespaceForTest, annotation.ProxyVisibility)).ExpectSuccess()
 				Expect(kubectl.CiliumEndpointWaitReady()).To(BeNil())
 
-				checkProxyRedirection(app1PodIP, false)
+				checkProxyRedirection(resource, false, parserType)
+			}
+
+			It("Tests HTTP proxy visibility without policy", func() {
+				proxyVisibilityTest(app1PodIP, app1Pod, "<Ingress/80/TCP/HTTP>", policy.ParserTypeHTTP)
+			})
+
+			It("Tests DNS proxy visibility without policy", func() {
+				proxyVisibilityTest("http://github.com", app2Pod, "<Egress/53/UDP/DNS>", policy.ParserTypeDNS)
```

Comment:
I should deploy the `bind-deployment.yaml` as part of this test and use that instead to avoid connectivity issues in the CI. Will do that in a follow-up commit shortly.